### PR TITLE
Gun Safeties

### PR DESCRIPTION
**Pull request by rye-rice**
## About The Pull Request
Request from a friend

Adds gun safeties as I have seen people accidentally discharge their weapons one too many times

Adjusts tactical reload to 1 second rather than 1.2 seconds

Screenshot_7922

Screenshot_7923

overlay only appears in inventory
Screenshot_7924

## Why It's Good For The Game
~~Gun safety RP~~
Accidentally shooting someone happens far too many times to be just a 'skill issue'

## Changelog

:cl:
add: Adds gun safetys
balance: Tacitcal reload time adjusted from 1.2 seconds to 1 second
/:cl:

### CODE REVIEW

**Comment by Zevotech** [comment]
Could you make the click not audible/visible to nearby players? I know it might be like, a balance issue but I don't want my crewmate to know I nearly accidentally shot them while trying to examine them. Its awkward both IC and OOC.

**Comment by rye-rice** [comment]
Oh that's fair, ill go edit that then

Though that removes opportunities to scream at the miner for bad gun safety

**Comment by thgvr** [comment]
Dunno, I think the sound being audible is good

**Comment by rye-rice** [comment]
im pretty sure they're refering to the *click* when a gun fails to fire, not the toggle saftey click

**Comment by thgvr** [comment]
That's kinda funny tbh

**Comment by Zevotech** [comment]
Its a tossup between it being IC'ly reactible and it being an OOC mistake someone makes. Most of the time when someone gets shot accidentally both players acknowledge it was a misclick or positioning mistake and shrug it off IC without much fanfare to it. I just. dont like the awkwardness of someone hearing the *click* as i'm looking at them and having to see the internal calculation of "did he just try to shoot me or was it a mistake and should i acknowledge it IC"

**Comment by tmtmtl30** [comment]
I _believe_ the safety typically blocks the trigger from moving, so it shouldn't make a click round when you attempt to fire with the safety on.

**Comment by thgvr** [comment]
I'm no gun nut but I'm fairly certain it varies

**Comment by thgvr** [comment]
Also, incredible, it fails the projectile unit test because the gun can't fire from having a safety on. Could be worth editing the unit test to disable the safety before firing?

**Comment by rye-rice** [comment]
LMAO, i think i can fix it, maybe

**Comment by Sun-Soaked** [comment]
> That's kinda funny tbh

From a mechanical standpoint, you should ALWAYS get auditory feedback on your gun failing to fire. If it just happens in the forest  without making a sound, it feels like a misinput and can get clunky quickly, especially if it's in a combat sitch

**Comment by thgvr** [comment]
Good to go?

**Comment by rye-rice** [comment]
Yes it is ready more or less

**Comment by thgvr** [comment]
Will review and merge tomorrow

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -198,6 +204,9 @@
 	else if(can_bayonet)
 		. += "It has a <b>bayonet</b> lug on it."
 
+	if(has_safety)
+		. += "The safety's [safety ? "<span class='green'>ON</span>" : "<span class='red'>OFF</span>"]. Ctrl-Click to toggle the safety."
```
Comment (MarkSuckerberg) [inline code/modules/projectiles/gun.dm]:
```suggestion
		. += "The safety is [safety ? "<span class='green'>ON</span>" : "<span class='red'>OFF</span>"]. Ctrl-Click to toggle the safety."
```

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -210,11 +219,16 @@
...
 /obj/item/gun/proc/can_shoot()
+	if(safety)
+		return FALSE
 	return TRUE
 
 /obj/item/gun/proc/shoot_with_empty_chamber(mob/living/user as mob|obj)
-	to_chat(user, "<span class='danger'>*[dry_fire_text]*</span>")		//WS Edit - Dry firing
-	playsound(src, dry_fire_sound, 30, TRUE)
+	if(!safety)
+		to_chat(user, "<span class='danger'>*[dry_fire_text]*</span>")		//WS Edit - Dry firing
```
Comment (MarkSuckerberg) [inline code/modules/projectiles/gun.dm]:
```suggestion
		to_chat(user, "<span class='danger'>*[dry_fire_text]*</span>")
```

File: code/modules/projectiles/gun.dm [hunk 3]
```diff
@@ -484,6 +498,25 @@
...
+
+	if(src == !user.get_active_held_item())
+		return
+
+	playsound(user, 'sound/weapons/gun/general/selector.ogg', 100, TRUE)
+	safety = !safety
+
+	user.visible_message(
+		span_notice("[user] turns the safety on the [src] [safety ? "<span class='green'>ON</span>" : "<span class='red'>OFF</span>"]."),
+		span_notice("You turn the safety on the [src] [safety ? "<span class='green'>ON</span>" : "<span class='red'>OFF</span>"]."),
```
Comment (MarkSuckerberg) [inline code/modules/projectiles/gun.dm]:
```suggestion
		span_notice("[user] turns the safety on [src] [safety ? "<span class='green'>ON</span>" : "<span class='red'>OFF</span>"]."),
		span_notice("You turn the safety on [src] [safety ? "<span class='green'>ON</span>" : "<span class='red'>OFF</span>"]."),
```

File: code/modules/projectiles/gun.dm [hunk 4]
```diff
@@ -669,6 +708,16 @@
...
 		. += knife_overlay
 
+	if(ismob(loc) && has_safety)
+		var/mutable_appearance/safety_overlay
+		safety_overlay = mutable_appearance('icons/obj/guns/safety.dmi')
+		switch(safety)
+			if(TRUE)
+				safety_overlay.icon_state = "safety-on"
+			if(FALSE)
+				safety_overlay.icon_state = "safety-off"
```
Comment (MarkSuckerberg) [inline code/modules/projectiles/gun.dm]:
```suggestion
		if(safety)
			safety_overlay.icon_state = "safety-on"
		else
			safety_overlay.icon_state = "safety-off"
```

**Review by MarkSuckerberg — DISMISSED** [review]

**Comment by thgvr** [comment]
I didn't review it but Mark did

**Pull request merged**